symengine/complex_mpc.cpp

```diff
@@ -41,9 +41,13 @@ int ComplexMPC::compare(const Basic &o) const
     const ComplexMPC &s = down_cast<const ComplexMPC &>(o);
     if (get_prec() == s.get_prec()) {
         int cmp = mpc_cmp(this->i.get_mpc_t(), s.i.get_mpc_t());
-        if (cmp == 0)
+        int x = MPC_INEX_RE(cmp), y = MPC_INEX_IM(cmp);
```

Comment:
presently in master 
```
    mpc_class c(100);
    mpc_pow_ui(c.get_mpc_t(), a.get_mpc_t(), 2, MPFR_RNDN);
    RCP<const Number> r1 = complex_mpc(std::move(c));

    mpc_class ee(100);
    mpc_set_ui_ui(ee.get_mpc_t(), 20, 14, MPFR_RNDN);
    RCP<const Number> r2 = complex_mpc(std::move(ee));

    std::cout<<r1>compare(*r2) << " , "<<r2->compare(*r1)<<"\n";
    
```

Output is `1 , 1`. # Atleast one should have been `-1`.
[MPC Docs](http://www.multiprecision.org/index.php?prog=mpc&page=html#index-mpc_005fcmp) mentions that the return value could be decomposed into `x` and `y` . and after decomposition , i tried to make compare of mpc close to the way we compare in [complex_double](https://github.com/symengine/symengine/blob/master/symengine/complex_double.cpp#L43-L53)